File: application/credit-card-enrollment-backend-in-java/src/main/java/cloud/ambar/common/ambar/httprequest/AmbarHttpRequest.java
package cloud.ambar.common.ambar.httprequest;

import cloud.ambar.common.serializedevent.SerializedEvent;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

@Data
@NoArgsConstructor
public class AmbarHttpRequest {
    @JsonProperty("data_source_id")
    @NonNull private String dataSourceId;
    @JsonProperty("data_source_description")
    @NonNull private String dataSourceDescription;
    @JsonProperty("data_destination_id")
    @NonNull private String dataDestinationId;
    @JsonProperty("data_destination_description")
    @NonNull private String dataDestinationDescription;
    @JsonProperty("payload")
    @NonNull private SerializedEvent serializedEvent;
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/Common/Ambar/AmbarHttpRequest.cs
using System.Text.Json.Serialization;
using CreditCardEnrollment.Common.Events;

namespace CreditCardEnrollment.Common.Ambar;

public class AmbarHttpRequest
{
    [JsonPropertyName("data_source_id")]
    public required string DataSourceId { get; init; }
    
    [JsonPropertyName("data_source_description")]
    public required string DataSourceDescription { get; init; }
    
    [JsonPropertyName("data_destination_id")]
    public required string DataDestinationId { get; init; }
    
    [JsonPropertyName("data_destination_description")]
    public required string DataDestinationDescription { get; init; }
    
    [JsonPropertyName("payload")]
    public required SerializedEvent SerializedEvent { get; init; }
}
